source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -265,10 +265,10 @@ void DnsResolverImpl::AddrInfoPendingResolution::onAresGetAddrInfoCallback(
     // Treat `ARES_ENODATA` or `ARES_ENOTFOUND` here as success to populate back the
     // "empty records" response.
     pending_response_.status_ = ResolutionStatus::Success;
-    pending_response_.details_ = "cares_norecords";
+    pending_response_.details_ = absl::StrCat("cares_norecords: ", ares_strerror(status));
     ASSERT(addrinfo == nullptr);
   } else {
-    pending_response_.details_ = "cares_failure";
+    pending_response_.details_ = absl::StrCat("cares_failure: ", ares_strerror(status));
```

Comment:
Hm, I'm pretty sure spaces are not allowed in details.
Can you verify that this is sanitized somewhere?  Any e2e test where this ends up in the logs is good enough - I'm 90% sure there ought to be an end to end test but that leaves a 10% chance of breaking something

Reply:
I updated the existing tests to assert that no empty spaces in the error details.